docs/BR.md

```diff
@@ -763,12 +770,20 @@ If a Random Value is used, the CA SHALL provide a Random Value unique to the Cer
   i. 30 days or
   ii. if the Applicant submitted the Certificate request, the time frame permitted for reuse of validated information relevant to the Certificate (such as in [Section 4.2.1](#421-performing-identification-and-authentication-functions) of these Guidelines or Section 11.14.3 of the EV Guidelines).
 
+CAs using this method MUST implement Multi-Perspective Issuance Corroboration as specified in Section 3.2.2.9. To count as corroborating, a Network Perspective MUST observe:
+- the same challenge information (i.e. Random Value or Request Token) as the Primary Network Perspective; and
+- evidence confirming the CA's authority to issue as determined by the Primary Network Perspective, as specified in Section 3.2.2.8. 
```

Comment:
3.2.2.9 (the new MPIC section) includes...

> Each corroborating Network Perspective MUST independently verify: 
a. the presence of the expected 1) Random Value, 2) Request Token, 3) IP Address, or 4) Contact Address, as required by the relied upon validation method specified in Sections 3.2.2.4 and 3.2.2.5; and
**b. the CA's authority to issue to the requested domain(s), as specified in Section 3.2.2.8.**

Given the bolded language above, it seems like we could remove the CAA references added in the 3.2.2.4.X subsections (e.g., "_evidence confirming the CA's authority to issue as determined by the Primary Network Perspective, as specified in Section 3.2.2.8._")?

Would that address your concern?

Reply:
I think there needs to be, at a minimum, a reference to section 3.2.2.9 in section 3.2.2.8. This reference would serve as an indicator to readers that they must also read 3.2.2.9 to correctly implement CAA checking.

Reply:
I think its worth having a more general discussion of this as well as a more detailed review of the language in the draft. Our group at Princeton feels that including some form of multi-perspective checking for CAA records is quite important. CAA records establish a CA's permission to issue and issuing in violation of a CAA record is a serious security breach.

However, we are aware there are some nuances regarding the implementation of CAA record checking in comparison to DV. Particularly, because CAA record checks must be done for all subdomains and have a shorter validity than a domain control validation check, CAA record checks may have to be done at issuance time for a domain that has already been validated. While it is clearly a stronger security stance and arguably a simpler standard to use multiple perspectives for all CAA checks, we felt it might meaningfully reduce the implementation burden to only perform MP-CAA checks at validation time (this is the current deployment at Let's Encrypt and I would be curious to also hear how Google Trust Services handles this case). Domain Validation contains more high-latency steps which depend on external systems than issuance does. To me this seems to facilitate an easier implementation.

As several have commented in the GitHub, only checking CAA records at validation time potentially raises several issues.

The first issue is the validity duration of a CAA check which is shorter than that of issuance. If MPCAA checking is only done at validation time, then this check would have to have validity equal to that of domain control validation to avoid a recheck at issuance time. Personally I think only performing a primary perspective CAA recheck at issuance time does not significantly change the security of the system. Potentially it allow an adversary to use a BGP attack to expand the time window after a malicious validation during which the adversary can request a certificate. However, this phenomena exists even with the shorter CAA checking window and other factors like DNS caching can cause this type of behavior even if checks were done more frequently. Personally I think this alone does not do much to justify rechecks at issuance time.

The second complication is that some validation methods can be used to validate arbitrary subdomains (while a CAA check must be done on each subdomain requested). This is a concern Dimitris brought up in the CAB Forum public list: https://lists.cabforum.org/pipermail/validation/2023-June/001904.html . In order to avoid all issuance time MPCAA checks, an MPCAA check would have to cover all subdomains. This is contrary to the CAA RFC that gives the CAA record of subdomains priority over parent domains. The vulnerability presented here is that an adversary can potentially downgrade the CAA record of a subdomain to that of its parent domain using an equally-specific BGP attack. This is clearly not ideal, but I still think that this is vastly preferable to dropping MPCAA record checking entirely. Also, in this scenario the adversary would be authorized to get a certificate for the parent domain (because it had completed domain control validation for that domain and the parent domain had a CAA record that permitted issuance) so this is largely about developing protections for a subdomain of a domain that an adversary has already compromised.

The final complication regards certificate types (i.e., wildcard vs standard) given that the interpretation of CAA records (particularly record sets with the issuewild tag) depends on the type of certificate a CA is signing (whereas some forms of domain validation can be used for either one). Potentially a CA may have checked a CAA record at validation time and been permitted to issue a standard, non-widlcard, certificate for that domain. At a later time a subscriber may request a wildcard certificate for the same domain. In some implementations this may require rechecking the CAA record to confirm that the CA is compliant with any issuewild tags. Personally I think there is potentially technical answer to this complication which is to store information about whether a CA can issue standard or wildcard certificate based on an initial CAA query (even if a subscriber is only requesting a standard certificate). One factor that facilitates this is that the same DNS query which is used to request the initial CAA record will both issuewild and issue tags. Storing this extra info would only require one additional bit of information on the result of an MPCAA lookup and would eliminate the need to perform additional MPCAA lookups at issuance time due to this reason.

Given all of these points and @CBonnell  's point about detaching the CAA checking requirement from the DV requirement, an idea that comes to mind is to add additional exceptions for the CAA checking requirement in 3.2.2.9:
Line 1054 could be modified to read:
[Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control) and [Section 3.2.2.5](#3225-authentication-for-an-ip-address) describe the validation methods that require the use of Multi-Perspective Issuance Corroboration and how a Network Perspective can corroborate the outcomes determined by the Primary Network Perspective. A CA MAY use corroborating evidence for CAA record compliance for a period of 398 days. If an applicant has completed domain control validation for a domain and a CA has collected corroborating evidence that gives the CA permission to issue a certificate for that domain as defined in Section 3.2.2.8 and 3.2.2.9, the CA does not need to corroborate the CAA determination of any subdomains for certificate requests from the same applicant.

Reply:
@CBonnell - Any opinion on @birgelee's suggestion above?

"_[Section 3.2.2.4](https://github.com/ryancdickson/staging/pull/8#3224-validation-of-domain-authorization-or-control) and [Section 3.2.2.5](https://github.com/ryancdickson/staging/pull/8#3225-authentication-for-an-ip-address) describe the validation methods that require the use of Multi-Perspective Issuance Corroboration and how a Network Perspective can corroborate the outcomes determined by the Primary Network Perspective. A CA MAY use corroborating evidence for CAA record compliance for a period of 398 days. If an applicant has completed domain control validation for a domain and a CA has collected corroborating evidence that gives the CA permission to issue a certificate for that domain as defined in Section 3.2.2.8 and 3.2.2.9, the CA does not need to corroborate the CAA determination of any subdomains for certificate requests from the same applicant._"